05-http-server-streams/02-file-server-post/server.js

```diff
@@ -1,16 +1,47 @@
 const url = require('url');
 const http = require('http');
 const path = require('path');
-
+const fs = require('fs-extra');
 const server = new http.Server();
+const LimitSizeStream = require('./LimitSizeStream');
 
-server.on('request', (req, res) => {
+server.on('request', async (req, res) => {
   const pathname = url.parse(req.url).pathname.slice(1);
 
   const filepath = path.join(__dirname, 'files', pathname);
 
   switch (req.method) {
     case 'POST':
+      res.statusCode = 201;
+      if (pathname.indexOf('/') !== -1) {
+        res.statusCode = 400;
+        return res.end();
+      } else if (await fs.exists(filepath)) {
```

Comment:
это очень и очень интересный момент: как именно определить есть ли уже файл на диске. если это выносить в отдельную операцию то это может привести к большой проблеме, т.к. файл может быть создан *между* вызовами `fs.exists()` и `fs.createWriteStream()`. правильнее здесь будет выполнить лишь 1 действие, т.е. только `fs.createWriteStream()`, однако по умолчанию такой вызов перезапишет файл. 
как же быть? ответ находится в опции под названием `flags`, которую можно передать в вызов этой функции (https://nodejs.org/dist/latest-v14.x/docs/api/fs.html#fs_file_system_flags).

также подробнее про потенциальную проблему с fs.exists можно прочитать тут: https://nodejs.org/dist/latest-v14.x/docs/api/fs.html#fs_fs_exists_path_callback
```
Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.
```